Assets/Tests/EditMode/AuthTests/FirebaseCredentialsProviderTests.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Moq;
+using NUnit.Framework;
+using Src.AuthController;
+using Src.AuthController.CredentialProviders.Google;
+using Src.AuthController.REST.REST_Response_DTOs;
+
+namespace Tests.EditMode.AuthTests
+{
+    public class FirebaseCredentialsProviderTests
+    {
+        [Test]
+        [TestCaseSource(nameof(GetAuthTypes))]
+        public async Task GetCredentialAsync_ShouldGetValidCredentials(FirebaseAuthProviderType firebaseAuthProviderType)
```

Comment:
Tests related to FirebaseCredentialProvider class should use this class. The problem with this class is that it takes responsibility of creating appropriate credentail provider on itself and hence it is problematic to determine whether or not it does this thing properly. Suggested solution: move logic related to creation of concrete credential providers into separate classes and use them via interfaces. By doing so you will be able to test the inner behaviour of FirebaseCredentialProvider by using mocks of created interfaces.